src/components/map/Map.tsx

```diff
@@ -78,6 +103,66 @@ export default function Map() {
         zoomOffset={-1}
         zIndex={mapZIndex.Labels}
       />
+      {state.showEstimatePins && state.explore.records.map((record, idx) => 
+        {
+          if(record.pin_latitude && record.pin_longitude){
+            return (
+              <Marker 
+                eventHandlers={{
+                  click: (e) => {
+                    (e.target as any).openPopup();
+                  },
+                }}
+                icon={record.estimate_grade === "National" ? nationalMarker : (record.estimate_grade === "Regional" ? regionalMarker : localMarker)}
+                key={`marker-${idx}`} 
+                position={[record.pin_latitude, record.pin_longitude]}
+              >
+                <Popup autoClose={false} className="pin-popup">
+                  <div className="popup-title">
+                    {`${record.estimate_grade} study details`}
+                  </div>
+                  <div className="popup-heading">
+                    Study name
+                  </div>
+                  <div className="popup-text">
+                    {`${record.source_name}`}
+                  </div>
+                  <div className="popup-heading">
+                    Location
+                  </div>
+                  <div className="popup-text">
+                    {record.city && `${record.city}`}{record.city && record.city.length !== 0 && record.state && record.state.length !== 0 && ", "}{record.state && `${record.state}`}{record.state && record.state.length !== 0 && record.country && ", "}{record.city && record.city.length !== 0 && record.state && record.state.length === 0 && record.country && ", "}{record.country && `${record.country}`}
```

Comment:
I think we might already have similar functionality here to compute "location text" (see https://github.com/serotracker/sero-can-webapp/blob/master/src/components/shared/StudiesTable.tsx#L66)

Let's consolidate both into a shared utility function for improved consistency/maintainability